### ISSUE TITLE
Projects that are related to K2

### ISSUE THREAD

**Issue by csukuangfj** [k2-fsa/k2#212]
I find that there are two projects, both from facebook, that have some overlaps with K2:

- [gtn][1], Dan mentioned this a few days ago
- [nestedtensor][2],  for tensors with irregular shapes, like `Ragged<T>`

Perhaps we can spend some time to find whether we can learn something from them.

[2]: https://github.com/pytorch/nestedtensor/
[1]: https://github.com/facebookresearch/gtn

**Comment by danpovey** [comment]
Yeah, cool.  I heard of gtn a couple days ago from several people.  Looks
interesting but seems to aim for CPU.

RE nestedtensor: the design seems quite different from TensorFlow's
RaggedTensor.  Much more general, but also much less similar
to what we are doing.  (TF has our row_ids and row_splits, which is a
design I came up with independently).

**Comment by awni** [comment]
Hi, (as Dan knows) I am one of the authors of GTN. 

I also started hearing about K2 a week or two ago from a few people :).

GTN is indeed currently CPU only with some vague plans at the moment to GPU accelerate some of the bottlenecks (namely compose). I find the GPU approach with K2 interesting. We are hoping to learn a bit more about your approach and how we can leverage K2 to do better work with differentiable WFSTs.

For reference here is a [preprint](https://arxiv.org/abs/2010.01003) that came out on arXiv today which describes the kinds of things we have started exploring with GTN and in general are interested in doing with a fast but easy to use differentiable WFST framework.  Maybe you will find it useful when designing your API.
